src/views/pools/util.test.ts

```diff
@@ -0,0 +1,178 @@
+      buyTxAverage: '0',
+      buyVolume: '0',
+      poolDepth: '400000',
+      poolFeeAverage: '0',
+      poolFeesTotal: '0',
+      poolROI: '0.5',
+      poolROI12: '0.5',
+      poolSlipAverage: '0',
+      poolStakedTotal: '300000',
+      poolTxAverage: '0',
+      poolUnits: '87500',
+      poolVolume: '0',
+      poolVolume24hr: '0',
+      price: '2',
+      runeDepth: '200000',
+      runeROI: '1',
+      runeStakedTotal: '100000',
+      sellAssetCount: '0',
+      sellFeeAverage: '0',
+      sellFeesTotal: '0',
+      sellSlipAverage: '0',
+      sellTxAverage: '0',
+      sellVolume: '0',
+      stakeTxCount: '2',
+      stakersCount: '1',
+      stakingTxCount: '2',
+      status: PoolDetailStatusEnum.Enabled,
+      swappersCount: '0',
+      swappingTxCount: '0',
+      withdrawTxCount: '0'
+    }
+    const priceIndex: PriceDataIndex = {
+      RUNE: bn(1),
+      FSN: bn(2)
+    }
+    it('returns PoolData for a FSN based pool', () => {
+      const expected: PoolDataType = {
+        pool: {
+          asset: 'RUNE',
+          target: 'FSN'
+        },
+        poolPrice: 'RUNE 2.000',
+        depth: 'RUNE 0.00',
+        volume: 'RUNE 0.00',
+        transaction: 'RUNE 0.00',
+        slip: '0',
+        trade: '0',
+        raw: {
+          depth: new BigNumber(200000),
+          volume: new BigNumber(0),
+          transaction: new BigNumber(0),
+          slip: new BigNumber(0),
+          trade: new BigNumber(0),
+          poolPrice: new BigNumber(2)
+        }
+      }
+      const result = getPoolData('RUNE', fsnPoolDetail, priceIndex, 'RUNE')
+      const rRaw = result.raw
+      const eRaw = expected.raw
+
+      expect(result.pool.asset).toEqual(expected.pool.asset)
+      expect(result.pool.target).toEqual(expected.pool.target)
+      expect(result.depth).toEqual(expected.depth)
+      expect(result.volume).toEqual(expected.volume)
+      expect(result.transaction).toEqual(expected.transaction)
+      expect(result.slip).toEqual(expected.slip)
+      expect(result.trade).toEqual(expected.trade)
+
+      expect(rRaw.depth).toEqual(eRaw.depth)
+      expect(rRaw.volume).toEqual(eRaw.volume)
+      expect(rRaw.transaction).toEqual(eRaw.transaction)
+      expect(rRaw.slip).toEqual(eRaw.slip)
+      expect(rRaw.trade).toEqual(eRaw.trade)
+      expect(rRaw.poolPrice).toEqual(eRaw.poolPrice)
+    })
+    it('returns PoolData for a BNB based pool', () => {
+      const expected: PoolDataType = {
+        pool: {
+          asset: 'RUNE',
+          target: 'BNB'
+        },
+        poolPrice: 'RUNE 0.000',
+        depth: 'RUNE 2.00',
+        volume: 'RUNE 0.00',
+        transaction: 'RUNE 0.00',
+        slip: '50100000',
+        trade: '1',
+        raw: {
+          depth: new BigNumber(199999799),
+          volume: new BigNumber(0),
+          transaction: new BigNumber(16193),
+          slip: new BigNumber(50100000),
+          trade: new BigNumber(1),
+          poolPrice: new BigNumber(0)
+        }
```

Comment:
When I see this test, I think `getPoolData` function should provide `raw` data only, but not formatted value of `raw` data. All formatted values can be easily created from `raw` data in a component by using format utility functions we have in `asgardex-util`. What do you think @FrozenIce0617 ?

Reply:
^ Moved into https://github.com/thorchain/asgardex-electron/issues/168